packages/backend-function/src/factory.ts

```diff
@@ -161,6 +162,10 @@ export type FunctionBundlingOptions = {
   minify?: boolean;
 };
 
+type DataResources = ResourceProvider<Record<string, unknown>> & {
+  modelIntrospectionSchema?: string;
+};
```

Comment:
I'm testing this against a modified version of `amplify-category-api`. Its unclear to me whether the MIS should be passed through at the root object level or within `.resources`.  Open to both, I left it at the top level to avoid additional plumbing, but I'm not sure what is best practice in this object.